### PR TITLE
Add Logger interface, AzureLoggerAdapter, and fix gRPC client stop

### PR DESCRIPTION
**Pull request by torosent**
# Summary

## What changed?
- Added `Logger` interface to standardize logging across the SDK
- Implemented `ConsoleLogger` for default logging to the console
- Implemented `NoOpLogger` for silent logging (useful in testing)
- Added `AzureLoggerAdapter` to integrate with Azure SDK logging infrastructure
- Added `ExponentialBackoff` class and utility functions (`sleep`, `withTimeout`) for retry scenarios
- Fixed `TaskHubGrpcClient.stop()` to properly await gRPC stub close

## Why is this change needed?
- The SDK needed a standardized logging interface to provide consistent logging behavior
- Azure SDK integration requires proper logging adapter for Azure-managed scenarios
- The gRPC client stop method needed to properly await cleanup to prevent premature termination

## Issues / work items
- Related to logging and reliability improvements

---

# Project checklist
- [ ] Release notes are not required for the next release
  - [ ] Otherwise: Notes added to `CHANGELOG.md`
- [ ] Backport is not required
  - [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
- [x] All required tests have been added/updated (unit tests, E2E tests)
- [ ] Breaking change?
  - [ ] If yes:
    - Impact:
    - Migration guidance:

---

# AI-assisted code disclosure (required)

## Was an AI tool used? (select one)
- [ ] No
- [x] Yes, AI helped write parts of this PR (e.g., GitHub Copilot)
- [ ] Yes, an AI agent generated most of this PR

If AI was used:
- Tool(s): GitHub Copilot
- AI-assisted areas/files: Logger implementations, backoff utilities, unit tests
- What you changed after AI output: Reviewed and integrated into codebase

AI verification (required if AI was used):
- [x] I understand the code and can explain it
- [x] I verified referenced APIs/types exist and are correct
- [x] I reviewed edge cases/failure paths (timeouts, retries, cancellation, exceptions)
- [x] I reviewed concurrency/async behavior
- [x] I checked for unintended breaking or behavior changes

---

# Testing

## Automated tests
- Result: New unit tests added for `AzureLoggerAdapter`, `ConsoleLogger`, `NoOpLogger`, and `ExponentialBackoff`

## Manual validation (only if runtime/behavior changed)
- Environment (OS, Node.js version, components): N/A - Unit tests cover functionality
- Steps + observed results: N/A
- Evidence (optional): N/A

---

# Notes for reviewers
- This PR introduces foundational logging infrastructure for the SDK
- The `AzureLoggerAdapter` enables proper integration with Azure SDK's logging system for the azure-managed package
- The backoff utility provides consistent retry behavior across the SDK

### CODE REVIEW

File: packages/durabletask-js-azuremanaged/src/azure-logger-adapter.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+ */
+export function createAzureLogger(namespace?: string): Logger {
+  const fullNamespace = namespace ? `durabletask:${namespace}` : "durabletask";
+  const azureLogger: AzureLogger = createClientLogger(fullNamespace);
+  
+  return {
+    error: (message: string, ...args: unknown[]): void => {
+      azureLogger.error(message, ...args);
+    },
+    warn: (message: string, ...args: unknown[]): void => {
```
Comment (YunchuWang) [inline packages/durabletask-js-azuremanaged/src/azure-logger-adapter.ts]: curious shall we align warn to warning to be same as azure logger?

File: submodules/durabletask-protobuf
[the remark this answers is not in the document]
Comment (torosent) [reply submodules/durabletask-protobuf]: fixed
